Robus/src/reception.c

```diff
@@ -54,11 +62,13 @@ void Recep_GetHeader(volatile uint8_t *data)
     // Catch a byte.
     MsgAlloc_SetData(*data);
     data_count++;
-
     // Check if we have all we need.
     switch (data_count)
     {
         case 1: //reset CRC computation
+#ifdef SNIFFER_H
+            timestamp[w_pos] = (uint64_t)LuosHAL_GetSystick() * 1000000;
+#endif /* SNIFFER_H */
```

Comment:
You timestamp the end of the first byte. meaning it's not really the beginning of the message because you already get 1 byte.
To make it right you should remove 1byte time from it : 
((uint64_t)LuosHAL_GetSystick() * 1000000) - (1/baudrate * 10);

Reply:
Yes, I timestamp at the end of the reception of the first byte but as the timestamp is neither really at ns nor at μs right now, it doesnt make such a difference for the level of frequencies we mainly use. If I decrement the number with the 1 byte time (which will be a few μs or less) we will end up with complex ts values (all ending with 999990 or sth like that) which will have the same relation as before, so for now I dont see the utility. I believe this is something to re-implement when we will have a real systick in μs.